lib/template-loader.php

```diff
@@ -0,0 +1,162 @@
+}
+add_action( 'wp_loaded', 'gutenberg_add_template_loader_filters' );
+
+/**
+ * Filters into the "{$type}_template" hooks to record the current template hierarchy.
+ *
+ * The method returns an empty result for every template so that a 'wp_template' post
+ * is used instead.
+ *
+ * @see gutenberg_find_template
+ *
+ * @param string $template  Path to the template. See locate_template().
+ * @param string $type      Sanitized filename without extension.
+ * @param array  $templates A list of template candidates, in descending order of priority.
+ * @return string Empty string to ensure template file is considered not found.
+ */
+function gutenberg_override_query_template( $template, $type, array $templates = array() ) {
+	global $_wp_current_template_hierarchy;
+
+	if ( ! is_array( $_wp_current_template_hierarchy ) ) {
+		$_wp_current_template_hierarchy = $templates;
+	} else {
+		$_wp_current_template_hierarchy = array_merge( $_wp_current_template_hierarchy, $templates );
+	}
+
+	return '';
+}
+
+/**
+ * Find the correct 'wp_template' post for the current hierarchy and return the path
+ * to the canvas file that will render it.
+ *
+ * @param string $template_file Original template file. Will be overridden.
+ * @return string Path to the canvas file to include.
+ */
+function gutenberg_find_template( $template_file ) {
+	global $_wp_current_template_post, $_wp_current_template_hierarchy;
+
+	// Bail if no relevant template hierarchy was determined, or if the template file
+	// was overridden another way.
+	if ( ! $_wp_current_template_hierarchy || $template_file ) {
+		return $template_file;
+	}
+
+	$slugs = array_map(
+		'gutenberg_strip_php_suffix',
+		$_wp_current_template_hierarchy
+	);
+
+	// Find most specific 'wp_template' post matching the hierarchy.
+	$template_query = new WP_Query(
+		array(
+			'post_type'      => 'wp_template',
+			'post_status'    => 'publish',
+			'post_name__in'  => $slugs,
+			'orderby'        => 'post_name__in',
+			'posts_per_page' => 1,
+		)
+	);
+
+	if ( $template_query->have_posts() ) {
+		$template_posts            = $template_query->get_posts();
+		$_wp_current_template_post = $template_posts[0];
+	}
+
+	// Add extra hooks for template canvas.
+	add_action( 'wp_head', 'gutenberg_viewport_meta_tag', 0 );
+
+	// This file will be included instead of the theme's template file.
+	return gutenberg_dir_path() . 'lib/template-canvas.php';
+}
+
+/**
+ * Renders the markup for the current template.
+ */
+function gutenberg_render_the_template() {
+	global $_wp_current_template_post;
+
+	if ( ! $_wp_current_template_post || 'wp_template' !== $_wp_current_template_post->post_type ) {
+		echo '<h1>' . esc_html__( 'No matching template found', 'gutenberg' ) . '</h1>';
+		return;
+	}
+
+	$content = $_wp_current_template_post->post_content;
+
+	$content = do_blocks( $content );
+	$content = wptexturize( $content );
+	$content = str_replace( ']]>', ']]&gt;', $content );
```

Comment:
I know this is in `the_content()` but why is it?

Reply:
I think it's because `wptexturize` doesn't handle that specific case?